pkg/asset/manifests/proxy.go

```diff
@@ -62,10 +62,27 @@ func (p *Proxy) Generate(dependencies asset.Parents) error {
 	}
 
 	if installConfig.Config.Proxy != nil {
+		noProxy := installConfig.Config.Proxy.NoProxy
+		if noProxy != "" {
+			if strings.Contains(noProxy, " ") {
+				return errors.New(fmt.Sprintf("noProxy name %s must not have spaces", noProxy))
+			}
+		}
+
+		httpProxy := installConfig.Config.Proxy.HTTPProxy
+		if !strings.HasPrefix(httpProxy, "http://") {
+			installConfig.Config.Proxy.HTTPProxy = fmt.Sprintf("%s%s", "http://", httpProxy)
```

Comment:
I don't think that we should add the scheme. For one thing, the https proxy does not need to use the https scheme. We should require the user to include the scheme.

I recommend that we do the same validation being done by the cluster-network-operator, with the exception of rejecting spaces in the noProxy rather than trimming them.
https://github.com/openshift/cluster-network-operator/blob/29c7414ae742621bcbcdc4d86a513f7927880dce/pkg/controller/proxyconfig/validation.go#L38